### PR TITLE
Unified configuration between GitPod and Codespaces using restructured utility scripts

### LINKED ISSUE #8: Fix dependency issues
- [ ] Protobuf https://github.com/lf-lang/examples-lingua-franca/issues/53
- [ ] Python-dev: [1]

[1]:

> Hello, I'm struggling to get the lingua franca playground going from here https://github.com/lf-lang/lingua-franca-playground. 
> I'm seeing messages as below when running via VSCode "Build and run" 
> ```
> /usr/bin/ld: /home/codespace/.python/current/lib/libpython3.10.a(bytearrayobject.o): relocation R_X86_64_PC32 against symbol `_Py_NoneStruct' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: bad value
> collect2: error: ld returned 1 exit status
> make[2]: *** [CMakeFiles/LinguaFrancahello.dir/build.make:195: /workspaces/lingua-franca-playground/examples/Python/src-gen/hello/LinguaFrancahello.so] Error 1
> make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/LinguaFrancahello.dir/all] Error 2
> make: *** [Makefile:136: all] Error 2
> ```
> 
> Running lfc on terminal gives me
> ```
> $ lfc hello.lf
> [100%] Linking C shared module /workspaces/lingua-franca-playground/examples/Python/src-gen/hello/LinguaFrancahello.so
> /usr/bin/ld: /home/codespace/.python/current/lib/libpython3.10.a(bytearrayobject.o): relocation R_X86_64_PC32 against symbol `_Py_NoneStruct' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: bad value
> collect2: error: ld returned 1 exit status
> make[2]: *** [CMakeFiles/LinguaFrancahello.dir/build.make:211: /workspaces/lingua-franca-playground/examples/Python/src-gen/hello/LinguaFrancahello.so] Error 1
> make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/LinguaFrancahello.dir/all] Error 2
> ```
> 
> Any suggestions?

### PR DESCRIPTION
**Pull request by axmmisaka**
This commit:
- Creates docker configs for Gitpod, GH Codespace and one for "LF general-use"
- Migrated GH Codespace to use DOCKERFILE config instead of installing after startup
- Added SDKMAN installation/config to all containers 
- Removed noexample configs as they are a bit redundant
- Potentially fixed and closes #8 (see pic)

TODOs:
- [ ] ......

image

### CODE REVIEW

File: dockerfiles/github-codespace.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM mcr.microsoft.com/devcontainers/universal:focal
+
+COPY .scripts .scripts
+RUN bash ./.scripts/setup.bash --dependencies-only &&\
+    # GitHub Codespaces container specific Java commands
+    bash -c "source /usr/local/sdkman/bin/sdkman-init.sh && sdk install java 19.0.2-open < /dev/null"
```
Comment (lhstrh) [inline dockerfiles/github-codespace.Dockerfile:6]: Shall we stick to Java 17 to ensure that the image is also suitable for building the LF compiler tool chain from source?

Comment (axmmisaka) [reply dockerfiles/github-codespace.Dockerfile:6]: Appears there's only `ms` version but I guess that's fine

Comment (lhstrh) [reply dockerfiles/github-codespace.Dockerfile:6]: What is an `ms` version?

Comment (axmmisaka) [reply dockerfiles/github-codespace.Dockerfile:6]: So there's a lot of build versions of openjdk, I was preferring the `open` version (that is, the "official" build of openjdk), but SDKMAN only has [Microsoft Built OpenJDK](https://www.microsoft.com/openjdk) now. After a second thought I don't think it's a big issue.

**Comment by axmmisaka** [comment]
Gitpod is having some very weird caching problem. Once it is fixed I think we can merge?

File: .scripts/setup-env.bash [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+INSTALL_USER_DEPENDENCIES=false
+
+for arg in "$@"; do
+    shift
+    case "$arg" in
+        '--install-user-dependencies') INSTALL_USER_DEPENDENCIES=true;;
+    esac
+done
+
+if [ "$INSTALL_USER_DEPENDENCIES" = true ]; then
```
Comment (axmmisaka) [inline .scripts/setup-env.bash:37]: Rationale of this change:
1. Gitpod appears to be complaining??? I think it's a GP issue 
2. We don't know if SDKMAN is installed locally or globally in the base image, if one exists. If it is installed locally it is not a good idea to do operations with them globally.

**Comment by axmmisaka** [comment]
It's working rn. Some stuff are not working with the base image and `dev` version, but I guess we could fix them in separate PRs as GH and GP configs are the main goal.

File: .scripts/setup-env.bash [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if (command -v "npm" &> /dev/null) ; then
+        echo "npm found"
+        npm install -g typescript
+    else
+        echo "npm not found, skipping TS installation. Recommend installing nvm as user."
+    fi
+
+    # Check if SDK is installed like what SDKMAN installer does
+    set +ux # GitPod will fail if this is not set. TODO: Investigate
```
Comment (lhstrh) [inline .scripts/setup-env.bash:46]:
```suggestion
```

**Review by lhstrh — APPROVED** [review]

**Pull request merged**